src/Tribe/Views/V2/Rest_Endpoint.php

```diff
@@ -16,6 +38,106 @@ class Rest_Endpoint {
 	 * @return  string
 	 */
 	public function get_url() {
-		return get_rest_url( null, Service_Provider::NAME_SPACE . '/html' );
+		if ( ! $this->is_available() ) {
+			$url = admin_url( 'admin-ajax.php' );
+			return add_query_arg( [ 'action' => static::$ajax_action ], $url );
+		}
+
+		return get_rest_url( null, static::NAMESPACE . '/html' );
+	}
+
+	/**
+	 * Register the endpoint if available.
+	 *
+	 * @since  TBD
+	 *
+	 * @return boolean If we registered the endpoint.
+	 */
+	public function register() {
+		if ( ! $this->is_available() ) {
+			return false;
+		}
+
+		return register_rest_route( static::NAMESPACE, '/html', [
+			'methods'             => Server::READABLE,
+			/**
+			 * @todo  Make sure we do proper handling of cache longer then 12h.
+			 */
+			'permission_callback' => static function ( Request $request ) {
+				return wp_verify_nonce( $request->get_param( '_wpnonce' ), 'wp_rest' );
+			},
+			'callback' => static function ( Request $request ) {
+				View::make_for_rest( $request )->send_html();
+			},
+			'args' => [
+				'url' => [
+					'required'          => true,
+					'validate_callback' => static function ( $url ) {
+						return is_string( $url );
+					},
+					'sanitize_callback' => static function ( $url ) {
+						return filter_var( $url, FILTER_SANITIZE_URL );
+					}
+				],
+			],
+		] );
+	}
+
+	/**
+	 * When REST is not available add ajax fallback into the correct action.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function maybe_enable_ajax_fallback() {
+		if ( $this->is_available() ) {
+			return;
+		}
+
+		$action = static::$ajax_action;
+		add_action( "wp_ajax_{$action}", [ $this, 'ajax_fallback' ] );
+		add_action( "wp_ajax_nopriv_{$action}", [ $this, 'ajax_fallback' ] );
+	}
+
+	/**
+	 * AJAX fallback for when REST endpoint is disabled. We try to mock a WP_REST_Request
+	 * and use the same method behind the scenes to make sure we have consistency.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function ajax_fallback() {
+		$request = new Request( 'GET', static::NAMESPACE . '/html' );
+
+		$body_params = (array) $_GET;
+		foreach ( $body_params as $key => $value ) {
+			$request->set_param( $key, $value );
+		}
+
+		View::make_for_rest( $request )->send_html();
+	}
+
+	/**
+	 * Check if the REST endpoint is available.
+	 *
+	 * @since  TBD
+	 *
+	 * @return boolean If the REST API endpoint is available.
+	 */
+	public function is_available() {
+		$is_available = tribe( 'tec.rest-v1.system' )->tec_rest_api_is_enabled();
```

Comment:
Should this check be for the WordPress REST API itself?
One could have deactivated TEC REST API but kept the WordPress REST API active.
The REST API we're using here is another matter entirely from the TEC REST API and we're filtering its availability below.

Reply:
Yeah I made a change to check what I could, but since 4.7 you cannot disable the rest API anymore, directly on WP. But we might need to have a internal HTTP request to do a real test of availability of the API.